counsel.el

```diff
@@ -5058,8 +5058,11 @@ When ARG is non-nil, ignore NoDisplay property in *.desktop files."
 (defun counsel--switch-buffer-unwind ()
   "Clear temporary file buffers and restore `buffer-list'.
 The buffers are those opened during a session of `counsel-switch-buffer'."
-  (mapc 'kill-buffer counsel--switch-buffer-temporary-buffers)
-  (mapc 'bury-buffer counsel--switch-buffer-previous-buffers)
+  (dolist (buf counsel--switch-buffer-temporary-buffers)
+    (kill-buffer buf))
```

Comment:
Why did you change the `mapc` to a `dolist`?

Reply:
Purely aesthetic, keeping similar things similar.  Maybe it would be better as:

    (mapc 'kill-buffer counsel--switch-buffer-temporary-buffers)
    (mapc 'bury-buffer (cl-remove-if-not
                        'buffer-live-p
                        counsel--switch-buffer-previous-buffers))

Reply:
Bonus points for properly quoting function symbols as `#'kill-buffer`, `#'bury-buffer`, etc. ;)

Reply:
Done and done!